experimental/telemetry/tracker.go

```diff
@@ -2,14 +2,44 @@ package telemetry
 
 import (
 	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-server/v6/plugin"
+
+	"github.com/mattermost/mattermost-plugin-api/experimental/bot/logger"
 )
 
+type TrackerConfig struct {
+	EnabledTracking bool
+	EnabledLogging  bool
+}
+
+// NewTrackerConfig returns a new trackerConfig from the current values from the plugin API.
+func NewTrackerConfig(api plugin.API) TrackerConfig {
```

Comment:
Passing a `plugin.API` makes the method hard to use with a https://pkg.go.dev/github.com/mattermost/mattermost-plugin-api#Client. Should `NewTrackerConfig` accept a `model.Config` instead?

Reply:
it's ok to me as long as we hide the complexity of the nil-zero-values logic inside this func. 👍🏽